### PR TITLE
Fix for Grafana breaking changes in v7.4 + minor bug fixes 

### PR DESCRIPTION
**Pull request by dhruv0000**
**Description**
- Fixed breaking changes by `Grafana v7.4`.
- UI will now check for previous `Grafana` and `Prometheus` config, and only auto connect if not configured.

PR Ref: #3159 
**Notes for Reviewers**
- Added a `GET` request for `/api/grafana/config` and `/api/Prometheus/config`

**[Signed commits](https://github.com/layer5io/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: handlers/grafana_handlers.go [hunk 1]
```diff
@@ -26,6 +26,17 @@ func (h *Handler) GrafanaConfigHandler(w http.ResponseWriter, req *http.Request,
 	// 	return
 	// }
 
+	if req.Method == http.MethodGet {
+		err := json.NewEncoder(w).Encode(prefObj.Grafana)
+		if err != nil {
+			logrus.Errorf("error marshaling Grafana config: %v", err)
```
Comment (leecalcote) [inline handlers/grafana_handlers.go:32]: We're ready for MeshKit errors. It's time to familiarize with new.error() and begin using MeshKit errors throughout Meshery's codebase.

Comment (leecalcote) [reply handlers/grafana_handlers.go:32]: @kumarabd did you make an attempt on the contributor instructions for this? // @DelusionalOptimist

File: router/server.go [hunk 2]
```diff
@@ -103,7 +103,7 @@ func NewRouter(ctx context.Context, h models.HandlerInterface, port int) *Router
 	gMux.Handle("/api/grafana/scan", h.ProviderMiddleware(h.AuthMiddleware(h.SessionInjectorMiddleware(h.ScanGrafanaHandler))))
 
 	gMux.Handle("/api/prometheus/config", h.ProviderMiddleware(h.AuthMiddleware(h.SessionInjectorMiddleware(h.PrometheusConfigHandler)))).
-		Methods("POST", "DELETE")
+		Methods("GET", "POST", "DELETE")
```
Comment (leecalcote) [inline router/server.go:106]: @piyushsingariya what do we need to do to keep the swagger definition and the REST API docs up to date when changes to the REST APIs are made?

File: ui/components/DashboardComponent.js [hunk 3]
```diff
@@ -237,6 +237,72 @@ class DashboardComponent extends React.Component {
...
+        self.props.updateProgress({ showProgress: false });
+        if (typeof result !== "undefined" && result?.grafanaURL && result?.grafanaURL !="") {
+          let selector = {
+            serviceMesh: "ALL_MESH",
+          };
+          fetchAvailableAddons(selector).subscribe({
+            next: (res) => {
+              res?.addonsState?.forEach((addon) => {
+                if (addon.name === "grafana" && ( self.state.grafanaURL === "" || self.state.grafanaURL == undefined )) {
+                  self.setState({grafanaURL: "http://" + addon.endpoint})
```
Comment (leecalcote) [inline ui/components/DashboardComponent.js:291]: No support for https?

File: router/server.go (hunk 2 above)
Comment (piyushsingariya) [reply router/server.go:106]: Just add the docs for new methods or endpoints.

**Pull request merged**